src/resmom/mom_hook_func.c

```diff
@@ -3517,7 +3518,7 @@ void reply_hook_bg(job *pjob)
 	return;
 
 	fini:
-		/*free(php->hook_output->reject_errcode);*/
+		free(php->hook_output->reject_errcode);
```

Comment:
Do you really need to free this? This is declared   in include/mom_hook_func.h as:
typedef struct mom_hook_output {
        int             *reject_errcode;
I

Reply:
We are doing malloc for this at below lines
[mom_server.c#L1101](https://github.com/PBSPro/pbspro/blob/master/src/resmom/mom_server.c#L1101) , [requests.c#L626](https://github.com/PBSPro/pbspro/blob/master/src/resmom/requests.c#L626)

Reply:
I see the malloc() part now. Thanks. So yes, it's okay to free() here.

Reply:
Now in mom_server.c, beginning in line 1094, we have this: 

					if ((phook_output = (mom_hook_output_t *)malloc(
						sizeof(mom_hook_output_t))) == NULL) {
							log_err(errno, __func__, MALLOC_ERR_MSG);
							goto err;
					}
					mom_hook_output_init(phook_output);

					if ((phook_output->reject_errcode =
						(int *)malloc(sizeof(int))) == NULL) {
							log_err(errno, __func__, MALLOC_ERR_MSG);
							goto err;
					}
If malloc() of reject_errcode fails, we also need to free up the malloc-ed phook_output in the previous statement. I looked at the 'err' part, and I don't see a line where phook_output is freed. And only when mom_process_hooks() is called, does the phook->output is put on a task.

Similar situation in requests.c just before the call to EXECJOB_END, we have:
619         if ((hook_output = (mom_hook_output_t *)malloc(
 620                 sizeof(mom_hook_output_t))) == NULL) {
 621                         log_err(errno, __func__, MALLOC_ERR_MSG);
 622                         return;
 623         }
 624         mom_hook_output_init(hook_output);
 625         if ((hook_output->reject_errcode =
 626                 (int *)malloc(sizeof(int))) == NULL) {
 627                         log_err(errno, __func__, MALLOC_ERR_MSG);
 628                         return;
 629         }
line 628 early return needs to free(hook_output)...

Reply:
Thank you Al, for catching this. I have addressed it.